### PR TITLE
added meme command

### PR DESCRIPTION
**Pull request by Progame03**
added a meme command, but this time I tried to do it properly

### CODE REVIEW

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URL;
+import java.util.Objects;
+
+public class MemeCommand extends SlashCommand {
+    public MemeCommand()
+    {
+        setCommandData(new CommandData("meme", "Gets a random random Meme from a specific subreddit")
+                .addOptions(new OptionData(OptionType.STRING, "subreddit", "Select the Subreddit you want to get the Meme from")
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:22]: This option needs to be required

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                        .addChoice("r/dankmemes", "dankmemes")
+                )
+        );
+    }
+
+    @Override
+    public void executeCommand(@NotNull SlashCommandEvent event, @Nullable Member sender, @NotNull SlashCommandContext ctx){
+        try
+        {
+            String requestURL = (!(event.getOption("subreddit")==null))?String.format("https://meme-api.herokuapp.com/gimme/%s", Objects.requireNonNull(event.getOption("subreddit")).getAsString()):"https://meme-api.herokuapp.com/gimme/memes";
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:35]:
```suggestion
            OptionMapping option = event.getOption("subreddit");
            String requestUrl = option == null
                ? "https://meme-api.herokuapp.com/gimme/memes"
                : String.format("https://meme-api.herokuapp.com/gimme/%s", option.getAsString())
```

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            LinkedDataObject json = LinkedDataObject.parse(url);
+            if (json == null)
+            {
+                ctx.replyError(ctx.getLocalized("commands.meme.api_down")).queue();
+                return;
+            }
+            EmbedBuilder builder = new EmbedBuilder()
+                    .setTitle(json.getString("title"))
+                    .setImage(json.getString("url"))
+                    .setDescription("\n\n[" + ctx.getLocalized("commands.meme.source") + "](" + json.getString("postLink") + ")")
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:46]: I don't think the newlines here actually work (Discord limitation)

File: src/main/resources/assets/languages/de.json [hunk 4]
```diff
@@ -163,6 +163,11 @@
 		},
 		"xp": {
 			"ranked" : "Hey {}, du bist gerade auf Level **{}** aufgestiegen!"
+		},
+		"meme":{
+			"api_down": "Die API kann momentan nicht erreicht werden. Bitte versuche es später erneut!",
+			"subreddit_not_existing": "Sieht aus als wär hier was schiefgelaufen!\nVersichere dich dass du deinen Subreddit korrekt geschrieben hast und versuche es erneut!",
```
Comment (Xirado) [inline src/main/resources/assets/languages/de.json:169]: You don't actually use this

File: src/main/resources/assets/languages/en_US.json [hunk 5]
```diff
@@ -163,7 +163,12 @@
 		},
 		"xp": {
 			"ranked" : "Hey {}, you just ranked up to level **{}**!"
-		}
+		},
+		"meme":{
+	        "api_down": "The API cannot be reached at the moment. Please try again later!",
+			"subreddit_not_existing": "Something went wrong here!\\nMake sure you spelled your subreddit correct and try again!\"",
```
Comment (Xirado) [inline src/main/resources/assets/languages/en_US.json:169]:
```suggestion
	    "api_down": "The API cannot be reached at the moment. Please try again later!",
			"subreddit_not_existing": "Something went wrong here!\nMake sure you spelled your subreddit correct and try again!",
```

**Review by Xirado — CHANGES_REQUESTED** [review]
Does the API filter NSFW posts?

**Comment by Progame03** [comment]
should've fixed everything

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                    ? "https://meme-api.herokuapp.com/gimme/memes"
+                    : String.format("https://meme-api.herokuapp.com/gimme/%s", option.getAsString());
+            URL url = new URL(requestURL);
+            LinkedDataObject json = LinkedDataObject.parse(url);
+            if (json == null)
+            {
+                ctx.replyError(ctx.getLocalized("commands.meme.api_down")).queue();
+                return;
+            }
+            if(!json.getBoolean("nsfw")){
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:46]:
```suggestion
            if (!json.getBoolean("nsfw"))
            {
```

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                EmbedBuilder builder = new EmbedBuilder()
+                        .setTitle(json.getString("title"))
+                        .setImage(json.getString("url"))
+                        .setDescription("[" + ctx.getLocalized("commands.meme.source") + "](" + json.getString("postLink") + ")")
+                        .setColor(0x152238);
+                ctx.reply(builder.build()).queue();
+            }
+            else{
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:54]:
```suggestion
            else
            {
```

File: src/main/resources/assets/languages/en_US.json [hunk 8]
```diff
@@ -163,7 +163,12 @@
 		},
 		"xp": {
 			"ranked" : "Hey {}, you just ranked up to level **{}**!"
-		}
+		},
+		"meme":{
+			"api_down": "The API cannot be reached at the moment. Please try again later!",
+			"source" : "Source",
+			"is_nsfw": "This is a NSFW Meme! In order for you to receive NSFW Memes your channel needs to be flagged as NSFW!"
```
Comment (Xirado) [inline src/main/resources/assets/languages/en_US.json:170]: You don't actually check if the channel is a NSFW channel

**Review by Xirado — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/languages/en_US.json (hunk 8 above)
Comment (Progame03) [reply src/main/resources/assets/languages/en_US.json:170]: jesus I'm braindead today

File: src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+                    : String.format("https://meme-api.herokuapp.com/gimme/%s", option.getAsString());
+            URL url = new URL(requestURL);
+            LinkedDataObject json = LinkedDataObject.parse(url);
+            if (json == null)
+            {
+                ctx.replyError(ctx.getLocalized("commands.meme.api_down")).queue();
+                return;
+            }
+
+            if (json.getBoolean("nsfw")&&!event.getTextChannel().isNSFW())
```
Comment (Xirado) [inline src/main/java/at/xirado/bean/command/slashcommands/MemeCommand.java:47]:
```suggestion
            if (json.getBoolean("nsfw") && !event.getTextChannel().isNSFW())
```

**Review by Xirado — CHANGES_REQUESTED** [review]
Press X to doubt

**Pull request merged**